### ISSUE TITLE
Add posts routes

### ISSUE THREAD

**Issue by alielbashir** [alielbashir/blog#4]
Posts router will include basic CRUD operations for blog posts

### Create post
POST /api/posts

#### example request
```json
{
    "title": "vscode tutorial",
    "content": "In this blog I'll be explaining how to use vscode, first step is to...",
}
```
#### example responses
```
HTTP status 201
HTTP status 401
HTTP status 403
```

### Get posts
GET /api/posts

#### example response
```json
{
    "posts": [
        {
            "title": "vscode tutorial",
            "id": 11,
            "creation_date": 1649212132,
            "username": "janedoe",
            "content": "In this blog I'll be explaining how to use vscode, first step is to...",
            "votes": 3
        },
        {
            "title": "codeblocks tutorial",
             "id": 131,
             "creation_date": 1649213057,
            "username": "johndoe",
            "content": "In this blog I'll be explaining how to use codeblocks, first step is to...",
            "votes": -26
        }
    ]
}
```

### Delete post
DELETE /api/posts/{id}

#### example response
```json
{"message": "post deleted successfully"}
```

### Upvote/Downvote
POST /api/posts/{id}/upvote

POST /api/posts/{id}/downvote

#### example response
```
HTTP status 200
HTTP status 404
```

**Comment by alielbashir** [comment]
for upvote and downvote, one thing I didn't take into account is duplicate votes. To prevent that, each post will have a field called voters with the IDs of every voter that voted on that post. Not sure if this is the best approach but it seems okay for the scale of the app

**Comment by alielbashir** [comment]
closed by #13
